multiqc/qualimap/__init__.py

```diff
@@ -0,0 +1,115 @@
+                    for r in os.listdir(os.path.join(root, 'raw_data_qualimapReport'))}
+                qualimap_raw_data[s_name]['plots'] = {os.path.splitext(r)[0]: os.path.join(root, 'images_qualimapReport', r) \
+                    for r in os.listdir(os.path.join(root, 'images_qualimapReport'))}
+
+        if len(qualimap_raw_data) == 0:
+            logging.debug("Could not find any QualiMap reports in {}".format(self.analysis_dir))
+            raise UserWarning
+
+        logging.info("Found {} QualiMap reports".format(len(qualimap_raw_data)))
+
+        self.sections = list()
+
+        # Section 1 - Coverage Histogram
+        histogram_data = self.qualimap_cov_his(qualimap_raw_data)
+        if len(histogram_data) > 0:
+            self.sections.append({
+                'name': 'Coverage Histogram',
+                'anchor': 'coverage-histogram',
+                'content': self.qualimap_cov_his_plot(histogram_data)
+            })
+
+
+    def qualimap_cov_his(self, qualimap_raw_data):
+        parsed_data = {}
+        for sn, data in qualimap_raw_data.iteritems():
+            cov_report = data['reports'].get('coverage_histogram')
+            if cov_report:
+                counts={}
+                try:
+                    with open(cov_report, 'r') as fh:
+                        next(fh) # skip the header
+                        for line in fh:
+                            (coverage, count) = line.split(None, 1)
+                            coverage = int(round(float(coverage)))
+                            count = float(count)
+                            counts[coverage] = count
+
+                except IOError as e:
+                    logging.error("Could not load input file: {}".format(cov_report))
+                    raise
+                parsed_data[sn] = counts
+
+        return parsed_data
+
+
+    def qualimap_cov_his_plot(self, parsed_data):
+        data = list()
+        for s in sorted(parsed_data):
+            pairs = list()
+            for k, p in iter(sorted(parsed_data[s].items())):
+                pairs.append([int(k), p])
+            data.append({
+                'name': s,
+                'data': pairs
+            })
+
+        html = '<div id="qualimap_cov_hist" class="hc-plot"></div> \n\
+                <script type="text/javascript"> \
+                    var cov_pconfig = {{ \n\
+                        "title": "Coverage Histogram",\n\
+                        "ylab": "Genome Bin Counts",\n\
+                        "xlab": "Coverage (X)",\n\
+                        "ymin": 0,\n\
+                        "xmin": 0,\n\
+                        "tt_label": "<b>{{point.x}}-X coverage </b>",\n\
+                        "use_legend": false,\n\
+                    }}; \n\
+                    $(function () {{ \
+                        plot_xy_line_graph("#qualimap_cov_hist", {d}, cov_pconfig); \
```

Comment:
I wish I could figure out a nice way to get the raw data out of the HTML :disappointed:

Not that it really matters I guess, but it just feels wrong!